### PR TITLE
Fix subclustering

### PR DESCRIPTION
**Pull request by boba-and-beer**
What subclustering does:

- Retrieves Documents from a **parent field**
- Gets those labels and attaches a "-" (this is used in the frontend to actually determine)
- Runs a for loop for each unique value in the parent field 
- Clusters on the filtered subset of the data 
- Adds a "-X" where X is the number for the subcluster to the unique value in parent field
- Updates documents with their relevant subclusters
- updates operation metadata

### CODE REVIEW

File: relevanceai/operations_new/manager.py [hunk 1]
```diff
@@ -21,25 +21,55 @@ def __init__(
...
+            ]
+            results = self.operation.insert_centroids(centroid_documents)
 
+        elif isinstance(self.operation, ClusterOps):
+            # If we get here, the operetion must be clustering and we should upsert centroid docs
             centroid_documents = self.operation.get_centroid_documents()
             self.operation.insert_centroids(centroid_documents)
 
+        for h in self.post_hooks:
+            h()
```
Comment (jtwinrelevanceai) [inline relevanceai/operations_new/manager.py:71]: nice

File: relevanceai/operations_new/ops.py [hunk 2]
```diff
@@ -515,3 +515,99 @@ def apply_transformers_pipeline(
...
+
+        # Building an infinitely hackable SDK
+
+        # Add filters and select fields
+        select_fields = vector_fields + [parent_field]
+        if filters is None:
+            filters = []
+
+        if cluster_ids is not None:
+            filters += [
```
Comment (jtwinrelevanceai) [inline relevanceai/operations_new/ops.py:556]: possible to use `self[parent_field] == cluster_id` here?

File: relevanceai/operations_new/run.py [hunk 3]
```diff
@@ -39,6 +39,10 @@ def run(
 
         from relevanceai.operations_new.manager import OperationManager
 
+        # store this
+        if hasattr(dataset, "dataset_id"):
+            self.dataset_d = dataset.dataset_id
```
Comment (jtwinrelevanceai) [inline relevanceai/operations_new/run.py:44]: `self.dataset_id`

**Review by jtwinrelevanceai — CHANGES_REQUESTED** [review]

File: relevanceai/operations_new/ops.py (hunk 2 above)
Comment (boba-and-beer) [reply relevanceai/operations_new/ops.py:556]: Yeah but using default filters just makes it slightly faster as it skips a layer of abstraction

File: relevanceai/operations_new/run.py (hunk 3 above)
Comment (jtwinrelevanceai) [inline relevanceai/operations_new/run.py:44]: its not changed?

**Review by jtwinrelevanceai — CHANGES_REQUESTED** [review]

**Comment by boba-and-beer** [comment]
Tested with Regine and frontend to make sure workflow worked. I think that's good enough for now?

**Pull request merged**